Update deploying_with_k8s.md with AMD ROCm GPU example

AlexHe99 opened the pull request:
Add the example of using AMD ROCm GPU

AlexHe99 committed:
Update deploying_with_k8s.md with AMD ROCm GPU example

Add the example of using AMD ROCm GPU

Signed-off-by: Alex He <alehe@amd.com>

DarkLight1337 commented:
Can we make each step a section and have separate sub-sections for NVIDIA vs AMD GPUs?

DarkLight1337 commented on docs/source/serving/deploying_with_k8s.md:
````diff
@@ -119,6 +119,79 @@ spec:
+        - name: shm
+          mountPath: /dev/shm
+```
+The full example is at https://github.com/ROCm/k8s-device-plugin/tree/master/example/vllm-serve.
````
```suggestion
The full example is at [https://github.com/ROCm/k8s-device-plugin/tree/master/example/vllm-serve].
```

Also we should make this an actual link.

AlexHe99 committed:
Update docs/source/serving/deploying_with_k8s.md

Good suggestion! Thank you.

Co-authored-by: Cyrus Leung <cyrus.tl.leung@gmail.com>
Signed-off-by: Alex He <alehe@amd.com>

AlexHe99 commented:
> To better organize things, can we make each step a section and have separate sub-sections for NVIDIA vs AMD GPUs?

AlexHe99 committed:
Update deploying_with_k8s.md

- Split it to two sub-section about the writing deployment.yaml for NVIDIA GPU and AMD GPU.

Signed-off-by: Alex He <alehe@amd.com>

AlexHe99 committed:
Update deploying_with_k8s.md

Signed-off-by: Alex He <alehe@amd.com>

AlexHe99 commented:
> To better organize things, can we make each step a section and have separate sub-sections for NVIDIA vs AMD GPUs?

I commit new patches to split the sections of deployment.yaml to two sub-sections. Other steps like PVC, SVC are same for both of them.

AlexHe99 committed:
Update deploying_with_k8s.md

Signed-off-by: Alex He <alehe@amd.com>

DarkLight1337 reviewed (approved):
This looks good to me, thanks for improving the documentation!
